Source/SpaceParkAPI/Controllers/PaymentsController.cs

```diff
@@ -0,0 +1,124 @@
+        }
+
+        
+
+        // GET all payments in the Database including parking.
+        [HttpGet]
+        public IActionResult GetAllPayments()
+        {
+            var query = (from p in _dbContext.Parkings
+                         join payment in _dbContext.Payments
+                             on p.Id equals payment.ParkId
+                         where p.Id == payment.ParkId
+                         select new
+                         {
+                             ParkingId = payment.Id,
+                             PersonName = p.UserName,
+                             Ship = p.SpaceShip,
+                             ArrivalTime = p.ArrivalTime,
+                             EndTime = payment.EndTime,
+                             Price = payment.Price
+
+                         }).ToList();
+
+            return Ok(query);
+        }
+
+        // GET payment by ParkId.
+        [HttpGet("{id}")]
+        public IActionResult GetPaymentByParkId(int id)
+        {
+            var parking =  _dbContext.Payments.Find(id);
+
+            if (parking == null)
+            {
+                return NotFound("That payment doesn't exist!");
+            }
+
+
+            var query = (from p in _dbContext.Parkings
+                         join payment in _dbContext.Payments
+                             on p.Id equals payment.ParkId
+                         where p.Id == id
+                         select new
+                         {
+                             ParkingId = payment.Id,
+                             UserName = p.UserName,
+                             Ship = p.SpaceShip,
+                             ArrivalTime = p.ArrivalTime,
+                             EndTime = payment.EndTime,
+                             Price = payment.Price
+
+                         }).ToList();
+
+           
+
+            return Ok(query);
+        }
+
+        // POST (and finish a parking) payment
+        [HttpPost]
+        public IActionResult PostPayment([FromBody] Pay pay)
+        {
+            
+                try
+                {
+                    var paidParking = _dbContext.Parkings.FirstOrDefault(p => p.Id == pay.ParkId);
+                    var currentSpacePort = _dbContext.SpacePorts.FirstOrDefault(s => s.Id == paidParking.SpacePortId);
+                    var parkingExist = _dbContext.Parkings.Any(p => p.Id == pay.ParkId);
+
+                    pay.EndTime = DateTime.Now;
+                    TimeSpan timeParked = (TimeSpan)(pay.EndTime - paidParking.ArrivalTime);
+
+                    pay.Price = timeParked.Minutes * 10;
+
+                    if (paidParking.Paid == true)
+                    {
+                        return BadRequest("The parking is already payed");
+                    }
+
+                    if (!parkingExist)
+                    {
+                        return BadRequest("There is no parking with this id");
+                    }
```

Comment:
```suggestion
                    if (paidParking.Paid == true)
                        return BadRequest("The parking is already payed");

                    if (!parkingExist)
                        return BadRequest("There is no parking with this id");
                    
                    pay.EndTime = DateTime.Now;
                    TimeSpan timeParked = (TimeSpan)(pay.EndTime - paidParking.ArrivalTime);

                    pay.Price = timeParked.Minutes * 10;
```

It seems like the `pay` variable isn't used within any of the if stmts.
From that understanding the if stmts may be moved above line 93.